roles/confluent.kafka_connect/files/kafka_connect.yml

```diff
@@ -1,38 +1,138 @@
+  # If you care about Producer/Consumer metrics for Connect, please uncomment 2 lines below.
+  # Please note that this increases the scrape duration by about 1-2 seconds as it needs to parse a lot of data.
+  - "kafka.consumer:*"
+  - "kafka.producer:*"
+blacklistObjectNames:
+  # This will ignore the admin client metrics from KSQL server and will blacklist certain metrics
+  # that do not make sense for ingestion.
+  - "kafka.admin.client:*"
+  - "kafka.consumer:type=*,id=*"
+  - "kafka.producer:type=*,id=*"
+  - "kafka.producer:client-id=confluent.monitoring*,*"
+  - "kafka.*:type=kafka-metrics-count,*"
 rules:
-  - pattern: 'kafka.connect<type=app-info, client-id=(.+)><>(.+): (.+)'
+  # "kafka.schema.registry:type=app-info,id=*"
+  - pattern: "kafka.connect<type=app-info, client-id=(.+)><>(.+): (.+)"
+    name: "kafka_connect_app_info"
     value: 1
-    name: kafka_connect_app_info
     labels:
-      client-id: $1
-      $2: $3
-    type: COUNTER
-  - pattern: 'kafka.connect<type=connector-metrics, connector=(.+)><>(.+): (.+)'
-    value: 1
-    name: kafka_connect_connect_connector_metrics
+      client-id: "$1"
+      $2: "$3"
+    type: UNTYPED
+  # kafka.connect:type=connect-worker-rebalance-metrics
+  - pattern: "kafka.connect<type=connect-worker-rebalance-metrics><>([^:]+)"
+    name: "kafka_connect_connect_worker_rebalance_metrics_$1"
+  # kafka.connect:type=connect-coordinator-metrics,client-id=*
+  # kafka.connect:type=connect-metrics,client-id=*
+  - pattern: "kafka.connect<type=(.+), client-id=(.+)><>([^:]+)"
+    name: kafka_connect_$1_$3
+    labels:
+      client_id: $2
+  # kafka.connect:type=connect-worker-metrics
+  - pattern: "kafka.connect<type=connect-worker-metrics><>([^:]+)"
+    name: kafka_connect_connect_worker_metrics_$1
+    labels:
+      connector: "aggregate"
+  # kafka.connect:type=connect-worker-metrics,connector=*
+  - pattern: "kafka.connect<type=connect-worker-metrics, connector=(.+)><>([^:]+)"
+    name: kafka_connect_connect_worker_metrics_$2
     labels:
       connector: $1
-      $2: $3
-    type: COUNTER
-  - pattern: "kafka.connect<type=connect-worker-metrics>([^:]+):"
-    name: "kafka_connect_connect_worker_metrics_$1"
-  - pattern: "kafka.connect<type=connect-metrics, client-id=([^:]+)><>([^:]+)"
-    name: "kafka_connect_connect_metrics_$1_$2"
-  - pattern: "kafka.connect<type=connector-task-metrics, connector=([^:]+), task=([^:]+)><>status: ([^:]+)"
-    name: "kafka_connect_connect_connector_metrics"
+  # kafka.connect:type=connector-metrics,connector=*
+  - pattern: "kafka.connect<type=connector-metrics, connector=(.+)><>(.+): (.+)"
     value: 1
+    name: kafka_connect_connector_metrics
     labels:
       connector: $1
-      task: $2
-      status: $3
-  - pattern: 'kafka.connect<type=(.+), connector=([^:]+), task=(\d+)><>([^:]+)'
-    name: kafka_connect_$1_$4
+      $2: $3
+    type: UNTYPED
+  # kafka.connect:type=*-task-metrics,*
```

Comment:
Should the commented out code get deleted?

Reply:
Nope, the comment identifies which JMX rule scrapes what JMX MBean. I created it highly detailed for promoting ease of use for people who do not know what those files are but still somehow get their scrape conditions and debugs sorted.

Reply:
Are the commented out lines are what you will see in prometheus? Even though they don't actually get used by the jmx exporter

Reply:
Not really. Prometheus format is built by rule below the comment. The comment suggests JVM MBean name (source) that will be scraped by the following rule/pattern. More like - This is the JMX metric being scraped by below rule.

Reply:
Roger that, thanks for the clarification